### ISSUE TITLE
Video and audio output selection within the app

### ISSUE THREAD

**Issue by JBraddockm** [rse/vingester#5]
Hi,

Thank you for this beautifully crafted application. I've been helping a friend remotely in setting his live stream setup and this application has become quite useful in bringing guess. However, unlike many people, we don't use OBS Ninja via OBS or NDI input via another software so this application's NDI output, as well as window capture aspect, don't apply to us so what I am about to say should kindly be considered in this context.

We use ATEM Mini Pro to bring in the guess via upstream key and use OBS for graphics only. I was wondering if it is possible to add a video output option where I could choose a connected external display onto which I can project OBS Ninja output as a fullscreen image. I don't know the exact terminology for this feature but I sometimes use QLAB and it has this projection feature where you choose the display and it projects contents onto it without the user having to move the actual window to the display and make it fullscreen. 

I think this could be useful in our case, not least because if for whatever reasons we have to close the stream and restart again, we can quickly re-project the stream again onto the same display rather than trying to find the display in the ATEM Mini screen and then try to make it fullscreen.

Similarly, if it is again possible, an audio output option could be quite useful. We also bring in guest audio into ATEM via HDMI. Currently, we use Sound Source to redirect the audio, which works fine and also remembers your choice. However, as it redirects all audio coming through via Vingester, it is not possible to choose different audio outputs for multiple guests. We currently have only one instance of OBS Ninja running so this is not an issue at the moment but in the case of multiple guests, this could be a problem.

Thank you again. Much appreciated.

**Comment by rse** [comment]
For the "video output option" you are talkin about, I think this is already possible with Vingester: Just use the Frameless mode and a browser width/height the size of the external display and specify a target display with the additional field after the "@" character. The field takes specifies like "[+-]x,[+-]y", so "-1,0" targets the display left to your primary display, "+1,0" targets the display right to your primary display, etc. The positioning of your additional displays relative to your primary display should be done through the OS preferences/settings.

For the "audio output option": the Web Contents could be forces to use a particular audio output device. For instance, OBS Ninja has the "audiodevice" option for directing the audio to a particular device. Check if your particular Web Contents can be forced in a similar way. Vingestern cannot easily force this itself, but I'll check whether I perhaps can redirect the Web Contents audio to a particular device similar to what I'm doing to currently capture the audio for the NDI case. I'll check.

**Comment by JBraddockm** [comment]
Thank you for the prompt reply. I've put +1,0 in @ it works nicely. My apologies for missing this feature earlier. 

However, maybe because I am on a Retina iMac, I had to put 3840x2160 as the size to get a full-screen projection on a 1080p monitor, albeit with the menu bar as I have the option "Displays have separate Spaces" ticked in Mission Control, which means I still have to go fullscreen manually. Similarly, I had to put 5120x2880 for iMac's built-in display. 

I'll check the audiodevice parameter in OBS Ninja.

Thank you.
